include/drivers/can.h

```diff
@@ -167,34 +167,44 @@ struct can_filter {
  *
  */
 struct zcan_frame {
-	/** Indicates the identifier type (standard or extended)
-	 * use can_ide enum for assignment
-	 */
-	uint32_t id_type : 1;
+	/** Message identifier*/
+	uint32_t id      : 29;
```

Comment:
Technically bitfield types can only be `int`, `unsigned int`, and `bool`.  Zephyr doesn't pay much attention to this.